src/coreclr/jit/emitarm64.cpp

```diff
@@ -9079,7 +9080,7 @@ void emitter::emitIns_Call(EmitCallType          callType,
     emitThisByrefRegs = byrefRegs;
 
     // for the purpose of GC safepointing tail-calls are not real calls
-    id->idSetIsNoGC(isJump || emitNoGChelper(methHnd));
+    id->idSetIsNoGC(isJump || isNoGCframe || emitNoGChelper(methHnd));
```

Comment:
Instead of passing around `isNoGCframe`, we could have added a check inside `emitNoGChelper()` to ignore for TLS, but adding a check for TLS `methHnd` might not solve the purpose, because for xarch we pass a dummy `methodHandle == 1` and for arm64, it is the actual address.

Reply:
I started with a solution where a special/fake JIT helper would be used to specify that we are dealing with a TLS call to the native runtime. In a way that is right - this is a JIT helper, just provided by the native runtime. That did not work though, since on arm64 the `methodHandle` is not just a marker.  

I also realized that we may want to opt-out more calls from being safeponts in the future. Not for correctness, but for "we do not need them" reason. We make nearly every call a safe point, but some calls would never be stackwalked.
Like native calls (not talking about entire pinvoke here, but the actual calls to native methods - it is ok if they are safepoints, but not necessary - a thing to think about anyways,..)

So I switched to a parameter that allows to override the default. Currently it would only be used for TLS calls, but we could use it for other cases.

This is basically the reason why I am passing around a parameter.